src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -50,7 +54,7 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
                     throw new ArgumentException($"Please call {nameof(AzureAppConfigurationOptions)}.{nameof(AzureAppConfigurationOptions.Connect)} to specify how to connect to Azure App Configuration.");
                 }
 
-                provider = new AzureAppConfigurationProvider(client, options, _optional);
+                provider = new AzureAppConfigurationProvider(client, options, _environmentName, _environmentNameEnabled, _optional);
```

Comment:
Do we want to create an instance of AzureAppConfigurationProvider even if the environment doesn't match?
Can we do the environment variable checks here instead of doing them in the constructor of AzureAppConfigurationProvider?